sunpy/physics/differential_rotation.py

```diff
@@ -142,3 +141,200 @@ def solar_rotate_coordinate(coordinate, new_observer_time, new_observer_location
+
+    Returns
+    -------
+    arr : `~numpy.ndarray`
+        Array with values between `original.min()` and `original.max()` . Note
+        that the type of the original image is not guaranteed to be reproduced.
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from sunpy.physics.transforms.differential_rotation import _un_norm
+    >>> original = np.array([-1, 0, 1])
+    >>> normalised = np.array([0., 0.5, 1.])
+    >>> out = _un_norm(normalised, original)
+    >>> out
+    array([-1.,  0.,  1.])
+    """
+    level = 0 if original.min() > 0 else np.abs(original.min())
+    arr *= original.max() + level
+    arr -= level
+    return arr
+
+
+@u.quantity_input(dt=u.s)
+def _warp_sun_coordinates(xy, smap, dt):
+    """
+    Function that returns a new list of coordinates for each input coord.
+    This is an inverse function needed by the scikit-image `transform.warp`
+    function.
+
+    Parameters
+    ----------
+    xy :
+        Array from `transform.warp`
+    smap : `~sunpy.map`
+        Original map that we want to transform
+    dt : `~astropy.units.Quantity`
+        Desired interval to rotate the input map by solar differential rotation.
+
+    Returns
+    -------
+    xy2 : `~numpy.ndarray`
+        Array with the inverse transformation
+    """
+    # NOTE: The time is being subtracted - this is because this function
+    # calculates the inverse of the transformation.
+    rotated_time = smap.date - timedelta(seconds=dt.to(u.s).value)
+
+    # Calculate the hpc coords
+    x = np.arange(0, smap.dimensions.x.value)
+    y = np.arange(0, smap.dimensions.y.value)
+    xx, yy = np.meshgrid(x, y)
+    # the xy input array would have the following shape
+    # xy = np.dstack([xx.T.flat, yy.T.flat])[0]
+
+    # We start by converting the pixel to world
+    hpc_coords = smap.pixel_to_world(xx * u.pix, yy * u.pix)
+
+    # then diff-rotate the hpc coordinates to the desired time
+    rotated_coord = solar_rotate_coordinate(hpc_coords, rotated_time)
+
+    # To find the values that are behind the sun we need to convert them
+    # to HeliographicStonyhurst
+    findOccult = rotated_coord.transform_to(HeliographicStonyhurst)
+
+    with np.errstate(invalid='ignore'):
+        # and find which ones are outside the [-90, 90] range.
+        occult = np.logical_or(np.less(findOccult.lon, -90 * u.deg),
+                               np.greater(findOccult.lon, 90 * u.deg))
+
+    # NaN-ing values that move to the other side of the sun
+    rotated_coord.data.lon[occult] = np.nan * u.deg
+    rotated_coord.data.lat[occult] = np.nan * u.deg
+    rotated_coord.cache.clear()
+
+    # Go back to pixel co-ordinates
+    x2, y2 = smap.world_to_pixel(rotated_coord)
+    # Re-stack the data to make it correct output form
+    xy2 = np.dstack([x2.T.value.flat, y2.T.value.flat])[0]
+    # Returned a masked array with the non-finite entries masked.
+    xy2 = np.ma.array(xy2, mask=np.isnan(xy2))
```

Comment:
What is the motivation for returning a masked array rather than an array with NaNs in?